models/gif_cond_exp_multisynapse.h

```diff
@@ -0,0 +1,467 @@
+
+  Author: March 2016, Setareh
+  SeeAlso: pp_psc_delta, gif_cond_exp, gif_cond_exp, gif_cond_exp_multisynapse
+*/
+
+namespace nest
+{
+
+extern "C" int
+gif_cond_exp_multisynapse_dynamics( double, const double*, double*, void* );
+
+class gif_cond_exp_multisynapse : public Archiving_Node
+{
+
+public:
+  gif_cond_exp_multisynapse();
+  gif_cond_exp_multisynapse( const gif_cond_exp_multisynapse& );
+  ~gif_cond_exp_multisynapse();
+
+  /**
+   * Import sets of overloaded virtual functions.
+   * @see Technical Issues / Virtual Functions: Overriding, Overloading, and
+   * Hiding
+   */
+  using Node::handle;
+  using Node::handles_test_event;
+
+  port send_test_event( Node&, rport, synindex, bool );
+
+  void handle( SpikeEvent& );
+  void handle( CurrentEvent& );
+  void handle( DataLoggingRequest& );
+
+  port handles_test_event( SpikeEvent&, rport );
+  port handles_test_event( CurrentEvent&, rport );
+  port handles_test_event( DataLoggingRequest&, rport );
+
+
+  void get_status( DictionaryDatum& ) const;
+  void set_status( const DictionaryDatum& );
+
+private:
+  void init_state_( const Node& proto );
+  void init_buffers_();
+  void calibrate();
+
+  void update( Time const&, const long_t, const long_t );
+
+  // make dynamics function quasi-member
+  friend int
+  gif_cond_exp_multisynapse_dynamics( double, const double*, double*, void* );
+
+  // The next two classes need to be friends to access the State_ class/member
+  friend class RecordablesMap< gif_cond_exp_multisynapse >;
+  friend class UniversalDataLogger< gif_cond_exp_multisynapse >;
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Independent parameters of the model.
+   */
+  struct Parameters_
+  {
+
+    double_t g_L_;
+    double_t E_L_;
+    double_t V_reset_;
+    double_t Delta_V_;
+    double_t V_T_star_;
+    double_t lambda_0_; /** 1/ms */
+
+    /** Refractory period in ms. */
+    double_t t_ref_;
+
+    /** Membrane capacitance in pF. */
+    double_t c_m_;
+
+    /** We use stc and sfa, respectively instead of eta and gamma
+    (mentioned in the references). */
+
+    /** List of spike-triggered current time constant in ms. */
+    std::vector< double_t > tau_stc_;
+
+    /** List of spike-triggered current jumps in nA. */
+    std::vector< double_t > q_stc_;
+
+    /** List of adaptive threshold time constant in ms. */
+    std::vector< double_t > tau_sfa_;
+
+    /** List of adaptive threshold jumps in mV. */
+    std::vector< double_t > q_sfa_;
+
+    /** Time constants of synaptic currents in ms */
+    std::vector< double_t > tau_syn_;
+
+    /** External DC current. */
+    double_t I_e_;
+
+    /** type is long because other types are not put through in GetStatus */
+    std::vector< long > receptor_types_;
+    size_t num_of_receptors_;
+
+    /** Reversal potential of synapses in mV */
+    double_t E_ex_; //!< Excitatory reversal Potential in mV
+    double_t E_in_; //!< Inhibitory reversal Potential in mV
```

Comment:
OK. Please let me know the outcome of the discussion. If it is necessary, I will change it to an array of reversal potentials.